engine/dlib/src/dlib/http_client.cpp

```diff
@@ -1017,12 +1025,13 @@ if (sock_res != dmSocket::RESULT_OK)\
             r = DoDoRequest(client, response, path, method);
             if (r != RESULT_OK && r != RESULT_NOT_200_OK) {
 
+                response.m_CloseConnection = 1;
```

Comment:
The bug fix.
Since we don't know what state the connection is in at this point, we should just as well close it, so we get a fresh one during next request.

Reply:
Makes sense!